### ISSUE TITLE
Allow specifying rules as camelCase strings

### ISSUE THREAD

**Issue by ZimGil** [eslint/eslint#13851]
**The version of ESLint you are using:** 7.13.0

**The problem you want to solve**: 
When specifying rules in a JavaScript module, due to the fact that rules names are kebab-cased, we have to "quote" the rule names.
It would be nice if we could support camelCase or even snake_case the rule names to avoid the need to "quote" the rule names.

**Your take on the correct solution to problem.**
We can change this line:
https://github.com/eslint/eslint/blob/37a06d633d3669f0f43236141dc43465b8bc7ec5/lib/linter/linter.js#L1168

To something like:
```javascript
        const configuredRulesBase = Object.assign({}, config.rules, commentDirectives.configuredRules);
        const configuredRules = lodash.reduce(configuredRulesBase , (kebabCased, ruleOptions, ruleId) => {
            kebabCased[ruleId.replace(/([A-Z])/gu, "-$1").toLowerCase()] = ruleOptions;
            return kebabCased;
        }, {});
```

**Are you willing to submit a pull request to implement this change?**
I would love to

**Comment by nzakas** [comment]
Thanks for the suggestion. At this point, I'm not in favor of making this change. In general, we try to stick to just having one way to do everything. While it might be nice to avoid typing quotes, I don't think that's a good enough reason to make such a change.

**Comment by mdjermanovic** [comment]
I agree with @nzakas.

Also, this wouldn't help with plugin rules, e.g., `"import/no-unresolved"`.

**Comment by nzakas** [comment]
Closing, as we won’t be moving forward with this. As a side note, if you use a JS config file, you can create your own function to translate rule names on the fly.

**Comment by nzakas** [comment]
Closing, as we won’t be moving forward with this. As a side note, if you use a JS config file, you can create your own function to translate rule names on the fly.
